lib/krb5/get_cred.c

```diff
@@ -963,6 +963,7 @@ get_cred_kdc_referral(krb5_context context,
     krb5_creds **referral_tgts = NULL;  /* used for loop detection */
     int loop = 0;
     int ok_as_delegate = 1;
+    int want_tgt = krb5_principal_is_krbtgt(context, in_creds->server);
```

Comment:
This needs to be true IFF `krb5_principal_is_krbtgt(context, in_creds->server)` AND `in_creds->server->realm` is not the "referral realm", i.e., the empty string.